ensime_shared/ensime.py

```diff
@@ -1097,12 +925,18 @@ def client_for(self, config_path, quiet=False, create_classpath=False,
         if abs_path in self.clients:
             client = self.clients[abs_path]
         elif create_client:
-            launcher = EnsimeLauncher(self.vim, config_path)
-            client = EnsimeClient(self.vim, launcher, config_path)
+            launcher = EnsimeLauncher(self.vim, config_path, self.server_v2)
+            client = self.do_create_client(launcher, config_path)
```

Comment:
It looks like you could move the launcher instantiation into `do_create_client()` now, simplifying what's passed around.